### DISCUSSION TITLE
QZ isn't controlling Sole f85 (2023 model) treadmill

### DISCUSSION THREAD

**Discussion by Sixdraw** [cagnulein/qdomyos-zwift#2920] [Q&A]
Hello, I recently purchased the QZ app and I'm having issues with the app controlling my Sole f85 treadmill. The QZ app reads the speed and incline accurately but when using the app to adjust the speed, the speed won't increase or decrease. The incline will increase to 1 but will not increase or decrease after that adjustment. 

When using the Peloton app the auto incline works fine but the auto speed adjustments are not working. 

Hopefully I've described the issues accurately, and in the proper forum, this is my first post on Github. Thanks.

**Comment by cagnulein** [comment]
Hi for the auto speed enable the speed force under the treadmill options

For the inclination set the inclination step setting to 1 in the treadmill
options

Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 19 dic 2024 alle 18:17 Sixdraw ***@***.***> ha
scritto:

**Reply by Sixdraw** [reply]
Hello cagnulein, thank you for your support. I changed the inclination step setting to 1 as you suggested and incline is now working correctly.

I had the speed force setting under treadmill options enabled previously. It's still not working.  (I still have it enabled)

**Comment by cagnulein** [comment]
Ate you able to change the speed from the speed tile in qz?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 19 dic 2024 alle 19:06 Sixdraw ***@***.***> ha
scritto:

**Reply by Sixdraw** [reply]
No. After pressing start, the treadmill begins at 0.5 (mi/h), and won't adjust when pressing plus or minus from the speed tile.

**Comment by cagnulein** [comment]
That's the issue.are you able to change the speed from the sole app
directly? If not It means that it's locked due to safety reason

It happens on some country

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

.

Il giorno gio 19 dic 2024 alle 20:05 Sixdraw ***@***.***> ha
scritto:

**Reply by Sixdraw** [reply]
Using the Sole+ app, i am able to make incline adjustments in display mode but I can't find where I can make any speed adjustments. 

I emailed service@soletreadmills.com to see if there is a setting I need to enable. I haven't heard back from them yet.

Would that cause the problem I'm having with auto adjusting speed in the Peloton app? And would auto speed be a problem with Zwift also?

**Comment by cagnulein** [comment]
Yes all about the speed is blocked on the firmware if you don't see the
speed adjustment in the sole app. Unfortunately on some countries sole
applies this. You should probably need to load a different firmware but I
don't know how

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 19 dic 2024 alle 23:29 Sixdraw ***@***.***> ha
scritto:

**Reply by Sixdraw** [reply]
"Yes all about the speed is blocked on the firmware if you don't see the
speed adjustment in the sole app."

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud/

I think I may have to question that statement? I'll try to explain.

Using Sole's built-in programs, auto speed and auto incline both work fine, but when using the treadmill's console to access Sole's online workouts, auto speed and incline are not working with the online programs. 

When running a workout with the Sole+ app on my smartphone, I had the same results, no auto speed or auto incline, yet I could manually adjust the incline from the display mode tab in the app after I stopped the workout. 

I downloaded and started a free trial period with the Kinomap app on my smartphone and have different results. It's my understanding that Sole Fitness has a partnership with Kinomap and has the Kinomap app pre-installed on my F85 treadmill.  Whether I run a workout from the Kinomap app on the console or the Kinomap app on my smartphone, auto incline is working but auto speed isn't. Which suggests to me that the Sole app isn't the determining factor.

Being the Sole online programs and the Sole+ app workouts aren't using auto speed or incline, yet auto incline works with Kinomap apps, wouldn't that suggest that auto speed and incline is available but auto speed is not being utilized?  

I'm still waiting for a response from  my email to Sole Fitness.

**Comment by cagnulein** [comment]
What you wrote make me thing that Auto speed work for preloaded program on
the sole firmware because are indeed preloaded in the firmware.

Just to be sure: even on kinomap you can't change speed from the app
directly, right?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 20 dic 2024 alle 18:38 Sixdraw ***@***.***> ha
scritto:

**Reply by Sixdraw** [reply]
After exploring the Kinomap app,  I couldn't find any option to manually control the speed or the incline with the Kinomap app. Perhaps someone could point out where I would find those manual options if available?

**Comment by cagnulein** [comment]
On a treadmill session it's enabled by default. The fact you don't see them
it's because they removed because the protocol doesn't handle.
Let's wait the sole answer

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 21 dic 2024 alle 00:06 Sixdraw ***@***.***> ha
scritto:
